File: go/014.go
package main

const BOUND = 1000 * 1000

func main() {
	var max, which int
	for i := 1; i < BOUND; i++ {
		count := 1
		for j := uint64(i); j > 1; count++ {
			if j % 2 == 0 {
				j = j / 2
			} else {
				j = j * 3 + 1
			}
		}
		if count > max {
			max = count
			which = i
			//println("Max:", max, "starts at", which)
		}
	}
	println("Max:", max, "starts at", which)
}

File: r/014.r
#!/usr/bin/Rscript

GOAL <- 1e6

max   <- 1
which <- 1

for (i in 1:GOAL) {
    count <- 1
    j <- i
    while (j > 1) {
        count <- count + 1

        if (j %% 2 == 0) {
            j <- j / 2
        } else {
            j <- 3 * j + 1
        }
    }
    if (count > max) {
        max <- count
        which <- i
        ##cat("Max:", max, "starts at", which, "\n")
    }
}

cat(which, "\n")
